src/instrument.ts

```diff
@@ -16,6 +16,11 @@
 
 import { startTracing } from './tracing';
 import { startMetrics } from './metrics';
+import { startProfiling } from './profiling';
+import { getEnvBoolean } from './options';
 
+startProfiling({
+  enabled: getEnvBoolean('SPLUNK_PROFILER_ENABLED', false),
```

Comment:
The main issue is that both metrics and profiling need to be switched on via an environment variable and are exported as functions.

When calling `startProfiling` programmatically (i.e. advanced config), setting the `enabled` option would make things ugly.

```javascript
startProfiling({ serviceName: 'foo', enabled: true }); // ??? I'm already calling start, why do I need to set it enabled
startProfiling({ serviceName: 'foo' }); // That's the way it should work, no extra params necessary.
```
The second option is the one I opted for, so it would work when starting instrumentation via CLI options as well.

I agree the enabled flag should be removed and `instrument.ts` just checks the env var without passing the flag to `startProfiling`, hoisted the flag out in https://github.com/signalfx/splunk-otel-js/pull/401/commits/f03ca4bd7b086096e86e33309c8884ff3e7bd545